pkg/storage/s3/s3.go

```diff
@@ -77,47 +89,63 @@ func GetConfig(listers *regopclient.Listers) (*S3, error) {
 		if err != nil {
-			return nil, fmt.Errorf("unable to get cluster minted credentials %q: %v", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.CloudCredentialsName), err)
+			return "", fmt.Errorf("unable to get cluster minted credentials %q: %v", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.CloudCredentialsName), err)
 		}
 
-		if v, ok := sec.Data["aws_access_key_id"]; ok {
-			cfg.AccessKey = string(v)
-		} else {
-			return nil, fmt.Errorf("secret %q does not contain required key \"aws_access_key_id\"", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.CloudCredentialsName))
-		}
-		if v, ok := sec.Data["aws_secret_access_key"]; ok {
-			cfg.SecretKey = string(v)
-		} else {
-			return nil, fmt.Errorf("secret %q does not contain required key \"aws_secret_access_key\"", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.CloudCredentialsName))
+		sharedCredsFile, err = sharedCredentialsFileFromSecret(sec)
+		if err != nil {
+			return "", fmt.Errorf("failed to save shared secrets file: %v", err)
 		}
 	} else if err != nil {
-		return nil, err
+		return "", err
 	} else {
+		var accessKey, secretKey string
 		if v, ok := sec.Data["REGISTRY_STORAGE_S3_ACCESSKEY"]; ok {
-			cfg.AccessKey = string(v)
+			accessKey = string(v)
 		} else {
-			return nil, fmt.Errorf("secret %q does not contain required key \"REGISTRY_STORAGE_S3_ACCESSKEY\"", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.ImageRegistryPrivateConfigurationUser))
+			return "", fmt.Errorf("secret %q does not contain required key \"REGISTRY_STORAGE_S3_ACCESSKEY\"", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.ImageRegistryPrivateConfigurationUser))
 		}
 		if v, ok := sec.Data["REGISTRY_STORAGE_S3_SECRETKEY"]; ok {
-			cfg.SecretKey = string(v)
+			secretKey = string(v)
 		} else {
-			return nil, fmt.Errorf("secret %q does not contain required key \"REGISTRY_STORAGE_S3_SECRETKEY\"", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.ImageRegistryPrivateConfigurationUser))
+			return "", fmt.Errorf("secret %q does not contain required key \"REGISTRY_STORAGE_S3_SECRETKEY\"", fmt.Sprintf("%s/%s", defaults.ImageRegistryOperatorNamespace, defaults.ImageRegistryPrivateConfigurationUser))
+		}
+
+		sharedCredsFile, err = sharedCredentialsFileFromStaticCreds(accessKey, secretKey)
+		if err != nil {
+			return "", fmt.Errorf("failed to save shared secrets file: %v", err)
 		}
 	}
 
-	return cfg, nil
+	return sharedCredsFile, nil
 }
 
-func (d *driver) updateConfigAndGetCredentials() (accessKey, secretKey string, err error) {
-	cfg, err := GetConfig(d.Listers)
+// updateConfigAndGetCredentialsLocation will return the location of an AWS
+// credentials config file that the caller is responsible for cleaning up.
+func (d *driver) updateConfigAndGetCredentialsLocation() (string, error) {
```

Comment:
Let's split this function into two:

* updateConfig
* getCredentialsLocation

I think the only reason why we have it as a single function so that people don't forget to call `updateConfig` before using it. Perhaps it should be checked via tests.

That'll help us to get rid of the local type `S3`.